pkg/clients/device/device.go

```diff
@@ -83,7 +83,7 @@ func NewClient(ctx context.Context, credentials []byte, projectID string) (Clien
 	}
 	deviceClient := CredentialedClient{
 		Client:      client.Client.Devices,
-		PortsClient: client.Client.DevicePorts,
+		PortsClient: client.Client.DevicePorts, //nolint:staticcheck
 		Credentials: client.Credentials,
 	}
```

Comment:
@displague along with those new credentials sources, we have some helpers to extract them so you can automatically get support for things like putting credentials in the filesystem or env vars. Here is an example :) https://github.com/crossplane/provider-aws/blob/959cc5b741fd7c2dafa7e20cac75ab55ae2c79bb/pkg/clients/aws.go#L105

Reply:
I think I've got those changes applied. I noticed the TestConnect tests were no longer present in other providers. I updated them in this provider after copying some private error message consts from crossplane-runtime.